File: backend-springboot/src/main/java/com/secureops/backendspringboot/waf/entity/WafEvent.java
package com.secureops.backendspringboot.waf.entity;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "waf_events")
public class WafEvent {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    public Long getId() { return id; }

    private String method;
    public String getMethod() { return method; }
    public void setMethod(String method) { this.method = method; }

    private String path;
    public String getPath() { return path; }
    public void setPath(String path) { this.path = path; }

    private String reason;
    public String getReason() { return reason; }
    public void setReason(String reason) { this.reason = reason; }

    private String sourceIp;
    public String getSourceIp() { return sourceIp; }
    public void setSourceIp(String sourceIp) { this.sourceIp = sourceIp; }

    private LocalDateTime createdAt;
    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt (LocalDateTime createdAt) { this.createdAt = createdAt; }
}

File: backend-Go/main/api/model/waf_event.go
package model

import "time"

type WafEvent struct {
	ID        int64     `gorm:"primaryKey" json:"id"`
	Method    string    `gorm:"not null" json:"method"`
	Path      string    `gorm:"not null" json:"path"`
	Reason    string    `gorm:"not null" json:"reason"`
	SourceIP  string    `gorm:"column:source_ip" json:"sourceIp"`
	CreatedAt time.Time `gorm:"column:created_at" json:"createdAt"`
}

func (WafEvent) TableName() string {
	return "waf_events"
}
